doc/lag/Static Port Channel Design Document.md

```diff
@@ -0,0 +1,278 @@
+    * [1.1 Requirements](#11-requirements)
+      * [1.1.1 Functional Requirements](#111-functional-requirements)
+      * [1.1.2 Configuration and Management Requirements](#112-configuration-and-management-requirements)
+      * [1.1.3 Scalability Requirements](#113-scalability-requirements)
+      * [1.1.4 Warm Boot Requirements](#114-warm-boot-requirements)
+    * [1.2 Design Overview](#12-design-overview)
+      * [1.2.1 Basic Approach](#121-basic-approach)
+      * [1.2.2 Container](#122-container)
+      * [1.2.3 SAI Overview](#123-sai-overview)
+  * [2 Functionality](#2-functionality)
+    * [2.1 Target Deployment Use Cases](#21-target-deployment-use-cases)
+      * [2.1.1 Enterprise](#211-enterprise)
+      * [2.1.2 Data Center](#212-data-center)
+    * [2.2 Functional Description](#22-functional-description)
+  * [3 Design](#3-design)
+    * [3.1 Overview](#31-overview)
+      * [3.1.1 Teamd configuration](#311-teamd-configuration)
+      * [3.1.2 Team Manager](#312-team-manager)
+    * [3.2 DB Changes](#32-db-changes)
+      * [3.2.1 CONFIG DB](#321-config-db)
+      * [3.2.2 APP DB](#322-app-db)
+      * [3.2.3 STATE DB](#323-state-db)
+      * [3.2.4 ASIC DB](#324-asic-db)
+      * [3.2.5 COUNTER DB](#325-counter-db)
+    * [3.3 Switch State Service Design](#33-switch-state-service-design)
+      * [3.3.1 Orchestration Agent](#331-orchestration-agent)
+      * [3.3.2 Other Process](#331-other-process)
+    * [3.4 SyncD](#34-syncd)
+    * [3.5 SAI](#35-sai)
+    * [3.6 CLI](#36-cli)
+      * [3.6.1 Data Models](#361-data-models)
+      * [3.6.2 Configuration Commands](#362-configuration-commands)
+      * [3.6.3 Show Commands](#363-show-commands)      
+      * [3.6.4 Debug Commands](#364-debug-commands)      
+      * [3.6.5 Rest API Support](#365-rest-api-support)      
+  * [4 Flow Diagrams](#4-flow-diagrams)
+    * [4.1 Port Channel Creation](#41-port-channel-creation)
+    * [4.2 Port Channel Member Addition](#42_port-channel-member-addition)
+  * [5 Error Handling](#5-error-handling)
+  * [6 Serviceability and Debug](#6-serviceability-and-debug)
+  * [7 Warm Boot Support](#7-warm-boot-support)
+  * [8 Scalability](#8-scalability)
+  * [10 Unit Test](#9-unit-test)
+  * [11 References](#references)
+    * [Openconfig support for Port Channel](#openconfig-support-for-port-channel)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 10/31/2019  | Amitabha Sen       | Initial version                   |
+| 0.2 | 06/20/2022  | Kannan Selvaraj    | Updated load-balance round-robin  |
+|     |             |                    | show command display as STATIC    |
+|     |             |                    | CONFIG_DB flag is changed from    |
+|     |             |                    | static to on                      |
```

Comment:
What consideration do you change "static" to "on"?

Reply:
Hi,

Took the Cisco CLI concept here.
on - Always port will be bundled into the group.
active - Wait for LACP protocol transaction to bundle the port into the group.

https://community.cisco.com/t5/switching/lacp-or-not-channel-group-1-mode-passive-active-or-on/td-p/772834
https://www.cisco.com/c/en/us/td/docs/switches/datacenter/nexus9000/sw/7-x/interfaces/configuration/guide/b_Cisco_Nexus_9000_Series_NX-OS_Interfaces_Configuration_Guide_7x/configuring_port_channels.pdf  (page 18)

![image](https://user-images.githubusercontent.com/90027415/183556881-f482a370-a8d1-433e-ad72-bfe5bac6bd7c.png)

Thanks,
Kannan.S

Reply:
CLI command "config portchannel add PortChannel<number> --static=true"
It is recommended to keep the original "static" naming in Rev 0.1.

Reply:
Sure will change accordingly and update the doc.

Reply:
Agree with chenkelly.

By default, there are not portchannels present in the system. Any newly created portchannel is Dynamic by default unless we explicitly distinguish using an option while portchannel creation.
The 'on' keyword is mainly used in Cisco CLI while adding a member port to portchannel. We may use the 'on' keyword for KLISH CLI; but in CLICK , we can continue with the '--static=true' option.

And also, please consider using the field-value pair "static":"true" in the CONFIG_DB entry "PORTCHANNEL|PortChannel1" instead of "on":"true".

Reply:
please consider using the --static=true for CLI static LAG.

Reply:
Sure Chenkelly accepted